sql-plugin/src/main/java/com/nvidia/spark/rapids/RapidsHostColumnVectorCore.java

```diff
@@ -165,16 +186,19 @@ public final Decimal getDecimal(int rowId, int precision, int scale) {
         "Assert fetch decimal with its original scale " + scale + " expected " + (-cudfCv.getType().getScale());
     if (precision <= Decimal.MAX_INT_DIGITS()) {
       assert cudfCv.getType().getTypeId() == DType.DTypeEnum.DECIMAL32 : "type should be DECIMAL32";
-      return Decimal.createUnsafe(cudfCv.getInt(rowId), precision, scale);
+      return Decimal.createUnsafe(getInt(rowId), precision, scale);
     } else {
       assert cudfCv.getType().getTypeId() == DType.DTypeEnum.DECIMAL64 : "type should be DECIMAL64";
-      return Decimal.createUnsafe(cudfCv.getLong(rowId), precision, scale);
+      return Decimal.createUnsafe(getLong(rowId), precision, scale);
     }
 
   }
 
   @Override
   public final UTF8String getUTF8String(int rowId) {
+    if (isNullAt(rowId)) {
+      return UTF8String.fromBytes(new byte[]{0});
```

Comment:
?
```suggestion
      return UTF8String.fromBytes("".getBytes("UTF8"));
```

Reply:
Curious if this could simply return `null`, since the calling code shouldn't be examining the value if the entry is marked as null.